libraries/AP_GPS/AP_GPS_UAVCAN.cpp

```diff
@@ -113,16 +117,59 @@ AP_GPS_Backend* AP_GPS_UAVCAN::probe(AP_GPS &_gps, AP_GPS::GPS_State &_state)
                 backend->_detected_module = i;
                 AP::can().log_text(AP_CANManager::LOG_INFO,
                                  LOG_TAG,
-                                 "Registered UAVCAN GPS Node %d on Bus %d\n",
+                                 "Registered UAVCAN GPS Node %d on Bus %d as instance %d\n",
                                  _detected_modules[i].node_id,
-                                 _detected_modules[i].ap_uavcan->get_driver_index());
+                                 _detected_modules[i].ap_uavcan->get_driver_index(),
+                                 _state.instance);
+                snprintf(backend->_name, ARRAY_SIZE(backend->_name), "UAVCAN%u-%u", _detected_modules[i].ap_uavcan->get_driver_index()+1, _detected_modules[i].node_id);
+                _detected_modules[i].instance = _state.instance;
             }
             break;
         }
     }
     return backend;
 }
 
+bool AP_GPS_UAVCAN::backends_healthy(char failure_msg[], uint16_t failure_msg_len)
```

Comment:
How is this particularly different then the `is_configured` method we already have in a GPS backend? It seems like everything that this function does is mapped out by `is_configured` and `broadcast_configuration_failure_reason`. Why do we need a new set of methods for everybackend?

Reply:
It's actually calling a static method from UAVCAN backend here, as the issue is to be detected before any backend is even initialised.